Find a functionally equivalent code:android.widget.ProgressBar.tileify:COMMENT
Method Modifier: private     
Comment:/**
 * Converts a drawable to a tiled version of itself. It will recursively
 * traverse layer and state list drawables.
 */

@@ -21,34 +21,39 @@
         }
         return out;
     } else if (drawable instanceof BitmapDrawable) {
-        final Bitmap tileBitmap = ((BitmapDrawable) drawable).getBitmap();
+        final BitmapDrawable bitmap = (BitmapDrawable) drawable;
+        final Bitmap tileBitmap = bitmap.getBitmap();
         if (mSampleTile == null) {
             mSampleTile = tileBitmap;
         }
         final ShapeDrawable shapeDrawable = new ShapeDrawable(getDrawableShape());
         final BitmapShader bitmapShader = new BitmapShader(tileBitmap, Shader.TileMode.REPEAT, Shader.TileMode.CLAMP);
         shapeDrawable.getPaint().setShader(bitmapShader);
-        return (clip) ? new ClipDrawable(shapeDrawable, Gravity.LEFT, ClipDrawable.HORIZONTAL) : shapeDrawable;
+        // Ensure the tint and filter are propagated in the correct order.
+        shapeDrawable.setTintList(bitmap.getTint());
+        shapeDrawable.setTintMode(bitmap.getTintMode());
+        shapeDrawable.setColorFilter(bitmap.getColorFilter());
+        return clip ? new ClipDrawable(shapeDrawable, Gravity.LEFT, ClipDrawable.HORIZONTAL) : shapeDrawable;
     }
     return drawable;
 }

